src/javascript/crypto/e2e/openpgp/pgpmime/pgpmail.js

```diff
@@ -0,0 +1,143 @@
+        value: constants.Mime.MULTIPART_MIXED}]});
+    var textNode = rootNode.addChild({multipart: false,
+      optionalHeaders: [{name: constants.Mime.CONTENT_TYPE,
+        value: constants.Mime.PLAINTEXT}]});
+    textNode.setContent(content.body);
+
+    goog.array.forEach(content.attachments, function(attachment) {
+      var filename = attachment.filename;
+      var options = {multipart: false,
+        contentType: {type: constants.Mime.OCTET_STREAM},
+        optionalHeaders: [{name: constants.Mime.CONTENT_TRANSFER_ENCODING,
+          value: constants.Mime.BASE64}
+        ]};
+      var attachmentNode = rootNode.addChild(options, filename);
+      attachmentNode.setContent(attachment.content);
+    });
+  }
+  return rootNode.buildMessage();
+};
+
+
+/**
+ * Builds a serialized MIME tree for PGP-encrypted content, per RFC 3156.
+ * @return {string}
+ */
+e2e.openpgp.pgpmime.PgpMail.prototype.buildPGPMimeTree = function() {
+
+  // Build the top-level node
+  var rootNode = new pgpmime.MimeNode({multipart: true,
+    optionalHeaders: [
+      // Default content transfer encoding is 7bit, according to RFC 2045
+      {name: constants.Mime.CONTENT_TYPE,
+        value: constants.Mime.DEFAULT_ENCRYPTED_CONTENT_TYPE},
+      {name: constants.Mime.CONTENT_TRANSFER_ENCODING,
+        value: constants.Mime.SEVEN_BIT},
+      {name: constants.Mime.SUBJECT, value: this.content.subject},
+      {name: constants.Mime.FROM, value: this.content.from},
+      {name: constants.Mime.TO, value: this.content.to},
+      {name: constants.Mime.IN_REPLY_TO, value: this.content.inReplyTo},
+      {name: constants.Mime.MIME_VERSION,
+        value: constants.Mime.MIME_VERSION_NUMBER}]});
+  rootNode.setContent(this.preamble);
+
+  // Set the required version info
+  // We're currently treating this node as an attachment (by including a
+  // name field), rather than solely containing the version/control information.
+  var versionNode = rootNode.addChild({multipart: false,
+    optionalHeaders: [
+      {name: constants.Mime.CONTENT_TYPE,
+        value: constants.Mime.ENCRYPTED,
+        params: {'charset': 'utf-8', 'name': 'version.asc'}},
+      {name: constants.Mime.CONTENT_TRANSFER_ENCODING,
+        value: constants.Mime.SEVEN_BIT},
+      {name: constants.Mime.CONTENT_DESCRIPTION,
+        value: 'PGP/MIME Versions Identification'}]});
+  versionNode.setContent(constants.Mime.VERSION_CONTENT);
+
+  // Set the ciphertext
+  // Due to Gmail bug, we use constants.Mime.PLAINTEXT instead of
+  // constants.Mime.OCTET_STREAM for contentType
+  var contentNode = rootNode.addChild({multipart: false,
+    optionalHeaders: [
+      {name: constants.Mime.CONTENT_TYPE,
+        value: constants.Mime.PLAINTEXT,
+        params: {'charset': 'utf-8', 'name': 'encrypted.asc'}},
+      {name: constants.Mime.CONTENT_TRANSFER_ENCODING,
+        value: constants.Mime.SEVEN_BIT}]});
+  contentNode.setContent(this.content.body);
```

Comment:
I see. In that case, the comment in the constructor should say "encrypted content of the email" instead of "unencrypted content of the email" (assuming the constructor is meant to be called with the encrypted/signed content). Also I think that means you need to expose `buildMimeTree_` (not private).